Source/BugsnagDeviceWithState.m

```diff
@@ -7,7 +7,85 @@
 //
 
 #import "BugsnagDeviceWithState.h"
+#import "BugsnagCollections.h"
+#import "BugsnagLogger.h"
+
+NSNumber *BSGDeviceFreeSpace(NSSearchPathDirectory directory) {
+    NSNumber *freeBytes = nil;
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    NSArray *searchPaths = NSSearchPathForDirectoriesInDomains(directory, NSUserDomainMask, true);
+    NSString *path = [searchPaths lastObject];
+
+    NSError *error;
+    NSDictionary *fileSystemAttrs =
+            [fileManager attributesOfFileSystemForPath:path error:&error];
+
+    if (error) {
+        bsg_log_warn(@"Failed to read free disk space: %@", error);
+    } else {
+        freeBytes = [fileSystemAttrs objectForKey:NSFileSystemFreeSize];
+    }
+    return freeBytes;
+}
+
+@interface BugsnagDevice ()
++ (void)populateFields:(BugsnagDevice *)device
+            dictionary:(NSDictionary *)event;
+
+- (NSDictionary *)toDict;
+@end
+
+@interface BugsnagDeviceWithState ()
+@property (nonatomic, readonly) NSDateFormatter *formatter;
+@end
 
 @implementation BugsnagDeviceWithState
 
+- (instancetype)init {
+    if (self = [super init]) {
+        _formatter = [NSDateFormatter new];
+        _formatter.dateFormat = @"yyyy'-'MM'-'dd'T'HH':'mm':'ssZZZ";
+        _formatter.timeZone = [NSTimeZone timeZoneForSecondsFromGMT:0];
```

Comment:
`Bugsnag` has a `+ (NSDateFormatter *)payloadDateFormatter` method.  Is there any reason not to use that?  The tests also repeat formatter initialisation and could be simplified.

Reply:
Updated the implementation to use `[Bugsnag payloadDateFormatter]`. I've left the tests using a different formatter as that should hopefully catch the case if something ever goes wrong with the `payloadDateFormatter`.

Reply:
Late-to-the-game addenda on the evils of repeated date formatter instantiation can be found e.g. [here](https://gist.github.com/JoshuaSullivan/6c0d0134e66c60c6cd8c).